### ISSUE TITLE
Different predictions for customized ssd_inception network through readNetFromTensorflow

### ISSUE THREAD

**Issue by TBondat** [opencv/opencv#12537]
##### System information (version)

- OpenCV python =>3.4.3.18
- Operating System / Platform => Windows 7 64bits

##### Detailed description
I have customized a pretrained model provided with tensorflow, the ssd_inception_v2_coco_2018_01_28.
The only change was the number of classes, i have only 2 classes (rear and front vehicle).
i didn't have a valid config .pbtxt file, so i generated one from your tf_text_graph_ssd.py v3.4.3 routine.
`python tf_text_graph_ssd.py --input myrearfrontveh.pb --output myrearfrontveh.pbtxt --num_classes 2`

2018-09-14 18:15:28.651777: I T:\src\github\tensorflow\tensorflow\tools\graph_transforms\transform_graph.cc:264] Applying sort_by_execution_order

So i got the .pbtxt.  I use the same 2 scripts examples as your nice explanation page https://github.com/opencv/opencv/wiki/TensorFlow-Object-Detection-API#generate-a-config-file

When running the one via opencv
`cvNet = cv.dnn.readNetFromTensorflow("myrearfrontveh.pb","myrearfrontveh.pbtxt") etc...` 
I get different results, actually predictions with very low confidence and so no correct prediction whereas it works well when i use the model directly with tensorflow (4 rear vehicles captured on the image vs 0 via opencv).

What could be the problem?

I have stored the 2 files myrearfrontveh.pb network and the myrearfrontveh.pbtxt  [here](https://github.com/TBondat/test_tensorflow_in_opencv) in case it is needed

Using default model like in your tutorial ssd mobilenet worked well with opencv.
thanks for helping.

**Comment by dkurt** [comment]
@TBondat, using [object_detection.py](https://github.com/opencv/opencv/blob/master/samples/dnn/object_detection.py):
```
python object_detection.py --input frame000000000.jpg --model myrearfrontveh.pb --config myrearfrontveh.pbtxt --mean 127.5 127.5 127.5 --scale 0.00784  --width 300 --height 300 --thr 0.1
```
output:
![out](https://user-images.githubusercontent.com/25801568/45747169-a5f9ac00-bc0d-11e8-8043-34cf887f27e1.png)

**Comment by TBondat** [comment]
Ok interesting! i didn't know this function. What is the difference between object_detection.py and the way presented in the main github page?
`cvNet.setInput(cv.dnn.blobFromImage(img, size=(300, 300), swapRB=True, crop=False))
cvOut = cvNet.forward()`
except that with cvOut i can process the output boxes.

Is there one new function vs a deprecated one? or the problem just come from my input arguments? 
Sorry, it is not cristal clear for me about the conclusions i should get. :/

**Comment by dkurt** [comment]
@TBondat, It looks like your .pbtxt file generated by older version than presented in the latest state of OpenCV. Now it inserts mean and scale normalization into the model (Mul and Sub nodes at the beggining of a .pbtxt file).

**Comment by TBondat** [comment]
ok i see! then, adding the parameters manually in the callback,
`cvNet.setInput(cv.dnn.blobFromImage(img, scalefactor=0.007843,size=(300, 300),
    mean=(127.5,127.5,127.5),swapRB=True,crop=False)`
gave me the predictions!
More than on your plot actually, not sure why exactly. If the .pb is the same. Did you fine-tune the .pbtxt with the Mul, Sub nodes you mention? i didn't.
![capture](https://user-images.githubusercontent.com/34160721/45756901-5d98b900-bc22-11e8-84ad-e10313c81e50.JPG)

Anyway, i get exactly the same predictions now in tensorflow and via opencv! Thousand thanks for the help!

**Comment by dkurt** [comment]
@TBondat, Sure, I just missed an `--rgb` flag:
```
python object_detection.py --input frame000000000.jpg --model myrearfrontveh.pb --config myrearfrontveh.pbtxt --mean 127.5 127.5 127.5 --scale 0.00784  --width 300 --height 300 --thr 0.1 --rgb
```
![out](https://user-images.githubusercontent.com/25801568/45758496-67c0b480-bc2e-11e8-855d-e28e806f74b1.png)

Can we close an issue?
